src/platform_linux/platform_inline.h

```diff
@@ -110,12 +110,21 @@ platform_semaphore_destroy(platform_semaphore *sema)
    debug_assert(!err);
 }
 
+/*
+ * Ref: https://man7.org/linux/man-pages/man3/sem_init.3.html
+ * for choice of 'pshared' arg to sem_init().
+ */
 static inline void
 platform_semaphore_init(platform_semaphore *sema,
                         int                 value,
-                        platform_heap_id    UNUSED_PARAM(heap_id))
-{
-   __attribute__((unused)) int err = sem_init(sema, 0, value);
+                        platform_heap_id    heap_id)
+{
+   // If we are running with a shared segment, it's likely that we
+   // may also fork child processes attaching to Splinter's shmem.
+   // Then, use 1 => spinlocks are shared across process boundaries.
+   // Else, use 0 => spinlocks are shared between threads in a process.
+   __attribute__((unused)) int err =
+      sem_init(sema, ((heap_id == PROCESS_PRIVATE_HEAP_ID) ? 0 : 1), value);
```

Comment:
@rtjohnso - Investigate why we had to this business of 0 / 1. Can't we always initialize sem_init() the same way, irrespective of whether it's process or threaded-exec model.

Reply:
@rtjohnso : Went back and took a re-look at the [man page](https://man7.org/linux/man-pages/man3/sem_init.3.html) mentioned a few lines above in the comment. 

The documented interface & its semantic is:

```
If pshared has the value 0, then the semaphore is shared between
       the threads of a process, and should be located at some address
       that is visible to all threads (e.g., a global variable, or a
       variable allocated dynamically on the heap).

       If pshared is nonzero, then the semaphore is shared between
       processes, and should be located in a region of shared memory.
```

So, no, doesn't seem like we can unconditionally always use either 0 or 1, irrespective of threaded- v/s process-execution model.